### ISSUE TITLE
Fix `doci_case` benchmark results

### ISSUE THREAD

**Issue by lelemmen** [GQCG/GQCP#650]
**Describe a change that would improve the code base**
Apparently, the DOCI matvec benchmark now takes 12 seconds, but I recall it being around 6 seconds.

**Additional context**
Maybe this speed decline was introduced by `EigenproblemEnvironment` and we're just copying the unnecessarily?

**Comment by guacke** [comment]
@lelemmen @dariatols what do you think about adding checks for the maximum allowed time for each benchmark? In this case, the additional build step would check for DOCI matvec <= 6s. These checks would then be specifically set to the Github Actions environments and should only be run as tests on that infrastructure (pure 'benchmarks' on other systems without timing constraints).

**Comment by dariatols** [comment]
I'm not fully up to date with EigenproblemEnvironment, but I think it is indeed a good idea to introduce a time limit.

**Comment by lelemmen** [comment]
@guacke @dariatols Indeed, that would be wonderful. Maybe we could specify further discussion in #535.
